src/main/java/dev/railroadide/railroad/ide/images/ColorPalette.java

```diff
@@ -0,0 +1,158 @@
+package dev.railroadide.railroad.ide.images;
+
+import dev.railroadide.core.ui.RRButton;
+import dev.railroadide.core.ui.RRVBox;
+import dev.railroadide.core.ui.localized.LocalizedText;
+import dev.railroadide.railroad.Railroad;
+
+import dev.railroadide.railroad.ide.IDESetup;
+import javafx.event.ActionEvent;
+import javafx.geometry.Insets;
+import javafx.geometry.Pos;
+import javafx.scene.Node;
+import javafx.scene.control.CheckBox;
+import javafx.scene.control.Label;
+import javafx.scene.input.MouseButton;
+import javafx.scene.layout.*;
+import javafx.scene.paint.Color;
+import javafx.scene.shape.Rectangle;
+import javafx.scene.text.Text;
+import javafx.stage.FileChooser;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+
+public class ColorPalette extends RRVBox {
+    private final FlowPane swatchPane;
+    public static final int SWATCH_SIZE = 20;
+    private Color selectedColor;
+    private  Palette palette;
+    private final CheckBox edit;
+    private final RRButton openFile = new RRButton("railroad.ide.image_editor.open_file");
+    private final Text rightClickToRemove = new Text("Right Click To Remove A Color");

```

Comment:
```suggestion
    private final Text rightClickToRemove = new Text("Right click to remove a color");
```
shouldn't have a capital for every word really